torch/testing/_internal/faulty_rpc_agent/backend_registry.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+import torch.distributed as dist
+import torch.distributed.rpc as rpc
+import torch.distributed.distributed_c10d as dc10d
+from torch.distributed.rpc import constants as rpc_constants
+
+from caffe2.torch.testing._internal.faulty_rpc_agent import faulty_process_group_agent_pybind
+
+def _faulty_process_group_construct_rpc_backend_options_handler(
+    rpc_timeout,
+    init_method,
+    num_send_recv_threads,
+    messages_to_fail,
+    num_fail_sends,
+    **kwargs
+):
+    return faulty_process_group_agent_pybind.FaultyProcessGroupRpcBackendOptions(
```

Comment:
instead of naming it `faulty_process_group_agent_pybind`, maybe we can call it `torch.distributed.rpc._testing` after moved it to `torch/csrc/`?